syntax = "proto3";

import "google/protobuf/any.proto";
import "attackType.proto";
import "cardAttribute.proto";
import "gameInfo.proto";
import "playerInfo.proto";

message GameEvent {
    sint32 player_id = 1;
    google.protobuf.Any event = 2;
}

message EventPlayerJoined {
    PlayerInfo player_info = 1;
}

message EventPlayerLeft {}
message EventGameClosed {}

message EventGameInfo {
    GameInfo game_info = 1;
}

message EventDeckSet {
    sint32 player_id = 1;
    string deck = 2;
}

message EventPlayerReady {
    sint32 player_id = 1;
}

message EventStartTurn {}
message EventLevelUp {}
message EventRefresh {}
message EventDiscardDownTo7 {}
message EventDeclareAttack {
    sint32 stage_pos = 1;
    AttackType attack_type = 2;
}

message EventPlayableCards {
    repeated sint32 hand_pos = 1;
}

message EventSetCardAttr {
    sint32 card_pos = 1;
    string zone = 2;
    ProtoCardAttribute attr = 3;
    sint32 value = 4;
}

message EventSetCardState {
    sint32 stage_pos = 1;
    CardState state = 2;
}

message EventGameEnded {
    bool victory = 1;
}

message EventRuleActionChoice {}
